pkg/store/proxy_test.go

```diff
@@ -1025,6 +1025,10 @@ func TestProxyStore_SeriesSlowStores(t *testing.T) {
 			return
 		}
 	}
+
+	// Wait until the last goroutine exits which is stuck on time.Sleep().
+	// Otherwise, goleak complains.
+	time.Sleep(5 * time.Second)
```

Comment:
Yea, I think that might be actually something we do wrong. Why we have to wait? What is not cleaned  5 seconds after the request. I don't think we can afford those eventual leaks in processes that will make dozens or hundreds QPS 🤔 

I think this is connected to the comment I made around `frameCtx`?